### DISCUSSION TITLE
Multiple GoTrueClient instances detected in the same browser context

### DISCUSSION THREAD

**Discussion by maaajo** [supabase/supabase#16062] [Questions]
I'm creating a project with nextjs and supabase.

For an unknown reason, I'm getting the following warning in the console:
Multiple GoTrueClient instances detected in the same browser context. It is not an error, but this should be avoided as it may produce undefined behavior when used concurrently under the same storage key.

I can't seem to find what could be causing this,

This is how I initiate the supabase instance in _app.tsx:
```javascript
export default function App({
  Component,
  pageProps,
}: AppProps<{ initialSession: Session }>) {
  const [supabaseClient] = useState(() => createBrowserSupabaseClient());

  return (
    <>
      <style jsx global>
        {`
          :root {
            --font-inter: ${inter.style.fontFamily};
          }
        `}
      </style>
      <SessionContextProvider
        supabaseClient={supabaseClient}
        initialSession={pageProps.initialSession}
      >
        <ChakraProvider theme={theme}>
          <Layout>
            <Component {...pageProps} />
          </Layout>
        </ChakraProvider>
      </SessionContextProvider>
    </>
  );
}
```

and this is how I consume the instance in components:
```javascript
const ForgotPassword = () => {
  const toast = useToast();
  const supabase = useSupabaseClient();
...
}
```

Have you ever had an issue like that and can help me understand what I'm doing wrong?

**Comment by kangmingtay** [comment]
Hey @maaajo, i think this is happening because you're passing a function which creates a Supabase client into useState. When you pass the `supabaseClient` into the `SessionProviderContext`, it invokes the function to create the client every time that page is rendered. Is there a reason why you can't do `const supabaseClient = createBrowserSupabaseClient()` directly?
